catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -138,7 +190,23 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
             return new SourceResponseImpl(request, Collections.emptyList());
         }
 
-        return super.query(request);
+        SourceResponse registryQueryResponse = super.query(request);
+        for (Result singleResult : registryQueryResponse.getResults()) {
+            if (singleResult.getMetacard()
+                    .getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                    .getValue()
+                    .toString()
+                    .equals(registryId)) {
+                if (!remoteName.equals(singleResult.getMetacard()
```

Comment:
I don't think this would cause any issues, especially since we require the user to add a title when using the UI, however it isn't strictly enforced. Added a check to ensure we aren't passing back a null string.